### ISSUE TITLE
Unhandled error - how to catch?

### ISSUE THREAD

**Issue by zxlin** [vanthome/winston-elasticsearch#113]
We're getting the following error on our logger, we know this is because the Elastic server ran out of storage; however, we can't seem to find out a way to catch and handle this error gracefully. Please see code below. We can get the error on the `error` event, but it still bubbles up as unhandled. Any ideas on how to catch and handle this? Thanks in advance!

```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error [ERR_UNHANDLED_ERROR]: Unhandled error. ({
  type: 'index_create_block_exception',
  reason: 'blocked by: [FORBIDDEN/10/cluster create-index blocked (api)];'
})
    at DerivedLogger.emit (events.js:212:17)
    at DerivedLogger.EventEmitter.emit (domain.js:475:20)
    at DerivedLogger.transportEvent (/app/node_modules/winston/lib/winston/logger.js:625:12)
    at Elasticsearch.emit (events.js:228:7)
    at Elasticsearch.EventEmitter.emit (domain.js:475:20)
    at /app/node_modules/winston-elasticsearch/bulk_writer.js:100:26
    at Array.forEach (<anonymous>)
    at /app/node_modules/winston-elasticsearch/bulk_writer.js:98:17
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

```JavaScript
const elasticSearchLogger = new WinstonElasticsearch({ ... });
// next line fires, but it still bubbles up as unhandled exception. 
elasticSearchLogger.on('error', (e) => console.error(`attempting to catch: ${e}`)); ```

**Comment by szabi84** [comment]
The same question here...

**Comment by zxlin** [comment]
@vanthome can we get a response on this when you get a chance? Thank you very much!

**Comment by vanthome** [comment]
Should be fixed with 0.8.6. If not, please re-open.

**Comment by Razzeee** [comment]
I seem to still be seeing this. In combination with uncaughtException logging after a `circuit_breaking_exception` happened. Which does get logged fine in the `on` block of the logger config.

```
    process
        .on('uncaughtException', (error) => {
            LoggerInstance.LogFatal(error, 'Uncaught Exception thrown');
            LoggerInstance.End(() => {
                process.exit(1);
            });
        });
```
